ocean_provider/utils/util.py

```diff
@@ -109,29 +110,38 @@ def _generate(_response):
 
 
 def get_service_files_list(
-    service: Service, provider_wallet: LocalAccount, version="4.1.0"
+    service: Service, provider_wallet: LocalAccount, asset: Asset = None
 ) -> list:
-    if version == "4.0.0":
+    if asset is None:
+        version = "4.0.0"
+    else:
+        version = asset.version if asset.version else "4.0.0"
```

Comment:
or a one-liner: `version = asset.version if asset is not None and asset.version else "4.0.0"`. Although I would rather default to 4.1.0 (CURRENT_VERSION constant)